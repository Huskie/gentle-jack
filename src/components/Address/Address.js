import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'react-emotion';
import { withTheme } from 'emotion-theming';

const StyledAddress = Styled('div')`
    margin-bottom: ${props => props.theme.spacing.componentMargin};
    span {
        display: block;
    }
`;

const BaseAddress = ({ children }) => <StyledAddress>{children}</StyledAddress>;

BaseAddress.propTypes = {
    children: PropTypes.node
};

BaseAddress.defaultProps = {};

const Address = withTheme(BaseAddress);

export { Address };