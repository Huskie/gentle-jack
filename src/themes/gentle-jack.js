const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
