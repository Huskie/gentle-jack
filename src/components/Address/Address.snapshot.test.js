import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { Address } from './Address';
import { theme } from '../../themes/gentle-jack';

describe('Address snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Address>
          <span>Registered office address:</span>
          <span>272 Bath Street</span>
          <span>Glasgow, G2 4JR</span>
        </Address>
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
