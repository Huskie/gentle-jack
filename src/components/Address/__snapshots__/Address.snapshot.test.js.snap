// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address snapshots: component renders with defaults 1`] = `
.emotion-0 {
  margin-bottom: 16px;
}

.emotion-0 span {
  display: block;
}

<div
  className="emotion-0"
>
  <span>
    Registered office address:
  </span>
  <span>
    272 Bath Street
  </span>
  <span>
    Glasgow, G2 4JR
  </span>
</div>
`;
